core/model.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import os
+import re
+from maxfw.model import MAXModelWrapper
+
+logger = logging.getLogger()
+
+
+class ModelWrapper(MAXModelWrapper):
+    """Model wrapper for Keras models"""
+
+    MODEL_NAME = 'lm_1b'
+    MODEL_LICENSE = "Apache v2"
+    MODEL_META_DATA = {
+        'id': '{}'.format(MODEL_NAME.lower()),
+        'name': '{} TensorFlow Model'.format(MODEL_NAME),
+        'description': 'Generative language model trained on the One Billion Words data set',
+        'type': 'Generative Language Model',
+        'license': '{}'.format(MODEL_LICENSE)
+    }
+
+    def __init__(self):
+        pass
+
+    def read_text(self, text_data):
+        text = text_data.decode("utf-8")
+        return text
+
+    def predict(self, x):
+        # this model does not like punctuation touching characters
+        x = re.sub('([.,!?()])', r' \1 ', x)  # https://stackoverflow.com/a/3645946/
+
+        os.system("python lm_1b/lm_1b_eval.py --mode sample \
+                                     --prefix \"" + x + "\" \
+                                     --pbtxt assets/data/graph-2016-09-10.pbtxt \
+                                     --vocab_file assets/data/vocab-2016-09-10.txt  \
+                                     --ckpt 'assets/data/ckpt-*' \
+                                     --num_samples 1")
+
+        try:
+            txt_file = open("out.txt", "r")
```

Comment:
Is the text file huge?

Reply:
No, the text file is very small. If the command on line 33 fails the text file never gets created.

Reply:
Ok I see

Reply:
This this change the OOM error now looks like this:
``` 
* Serving Flask app "Model Asset Exchange Microservice" (lazy loading)
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
172.17.0.1 - - [12/Feb/2019 21:23:11] "GET /swagger.json HTTP/1.1" 200 -
172.17.0.1 - - [12/Feb/2019 21:23:11] "GET /model/metadata HTTP/1.1" 200 -
Recovering graph.
Recovering checkpoint assets/data/ckpt-*
2019-02-12 21:23:22.985227: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
Killed
Error generating a prediction, this is likely due to a lack of memory allocated to Docker
The minimum recommended resources for this model is 8 GB Memory and 4 CPUs
172.17.0.1 - - [12/Feb/2019 21:23:39] "POST /model/predict HTTP/1.1" 200 -
```
instead of:
```
[2018-07-28 03:07:36,563] ERROR in app: Exception on /model/predict [POST]
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/conda/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/opt/conda/lib/python3.6/site-packages/flask_restplus/api.py", line 319, in wrapper
    resp = resource(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/flask/views.py", line 88, in view
    return self.dispatch_request(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/flask_restplus/resource.py", line 44, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/flask_restplus/marshalling.py", line 136, in wrapper
    resp = f(*args, **kwargs)
  File "/workspace/api/model.py", line 58, in post
    preds = self.model_wrapper.predict(text)
  File "/workspace/core/tensorflow.py", line 29, in predict
    txt_file = open("out.txt", "r")
FileNotFoundError: [Errno 2] No such file or directory: 'out.txt'
```

Reply:
We should comment this in the code